### DISCUSSION TITLE
is it possible to custom transform without losing the default transform?

### DISCUSSION THREAD

**Discussion by iamyuu** [pveyes/htmr#142] [Q&A] [answered]
I have a case editor who can input some special tags on WYSIWYG when I'm trying to custom transform some tags, the other tags are not rendered. Check my code on [codesandbox](https://codesandbox.io/s/htmr-custom-transform-ly940u?file=/src/App.js)

**Comment by pveyes** [comment, marked as answer]
There's 2 issues in your codesandbox

1. HTML only support self-closing tags for limited number of elements, e.g: `img`. This means `<hello />` won't work, you can instead use `<hello name="Codesandbox"></hello>` but I'd recommend using [custom element naming convention](https://usefulangle.com/post/363/custom-element-valid-name#:~:text=While%20creating%20a%20new%20custom,an%20uppercase%20ASCII%20character%20anywhere.) for this, e.g: `x-hello` as it's a valid HTML

3. You need to handle text nodes when rendering with custom transform, you can check for undefined `props`, so your transform should be like this

```js
htmr(html, {
  transform: {
      _: (node, props, children) => {
        if (!props) { return node } // this is a text node

        if (node === "hello") {
          return <h1>Hello {props.name}</h1>;
        }

        return createElement(node, props, children);
      }
  }
})
```

**Reply by iamyuu** [reply]
it works, thanks!
